### PR TITLE
Added example DAG for MSSQL to Google Cloud Storage (GCS)

### PR DESCRIPTION
**Pull request by Bowrna**
Added example DAG for MSSQL to Google Cloud Storage (GCS)
PR relevant to [this](https://github.com/apache/airflow/issues/8280) issue 

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

File: airflow/providers/google/cloud/example_dags/example_mssql_to_gcs.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+from airflow.utils import dates
+
+GCS_BUCKET = os.environ.get("GCP_GCS_BUCKET", "example-airflow")
+FILENAME = 'test_file'
+
+SQL_QUERY = "USE airflow SELECT * FROM Country;"
+
+with models.DAG(
+    'example_mssql_to_gcs',
+    default_args=dict(start_date=dates.days_ago(1)),
```
Comment (uranusjr) [inline airflow/providers/google/cloud/example_dags/example_mssql_to_gcs.py]:
```suggestion
    start_date=None,
```

I believe we are moving away from unnecessary `default_args` and dynamic `start_data`.

Comment (josh-fell) [reply airflow/providers/google/cloud/example_dags/example_mssql_to_gcs.py:30]: Yes and we've been adding `catchup=False` in all example DAGs to ward off DagRun headaches for new users.

@uranusjr I could be wrong here but having `start_date=None` would throw an exception that the `start_date` parameter is missing right?

File: docs/apache-airflow-providers-google/operators/transfer/mssql_to_gcs.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+~~~~~~~~~~~~~~~~~~
+
+:class:`~airflow.providers.google.cloud.transfers.mssql_to_gcs.MSSQLToGCSOperator` allows you to upload
+data from Microsoft SQL Server database to GCS.
+
+Below is an example of using this operator to upload data to GCS.
+
+.. exampleinclude:: /../../airflow/providers/google/cloud/example_dags/example_mssql_to_gcs.py
+    :language: python
+    :dedent: 0
```
Comment (josh-fell) [inline docs/apache-airflow-providers-google/operators/transfer/mssql_to_gcs.rst]: This isn't needed. The default value here is 0.

File: docs/apache-airflow-providers-google/operators/transfer/mssql_to_gcs.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Microsoft SQL Server To Google Cloud Storage Operator
+======================================
```
Comment (josh-fell) [inline docs/apache-airflow-providers-google/operators/transfer/mssql_to_gcs.rst]:
```suggestion
Microsoft SQL Server To Google Cloud Storage Operator
=====================================================
```
The underline needs to match the length of the line it's underlining in Sphinx.

File: airflow/providers/google/cloud/example_dags/example_mssql_to_gcs.py (hunk 1 above)
Comment (uranusjr) [reply airflow/providers/google/cloud/example_dags/example_mssql_to_gcs.py]: Uh, yes you’re right. Don’t do `start_date=None`.

Comment (josh-fell) [reply airflow/providers/google/cloud/example_dags/example_mssql_to_gcs.py]: @Bowrna Any valid, static `start_date` value is fine. Also, @uranusjr's suggested change of not having the `start_date` in `default_args` is a good one!

**Comment by Bowrna** [comment]
do i have to fix any more issues or is there more suggestions to fix in this PR?

**Review by potiuk — APPROVED** [review]

**Pull request merged**